### ISSUE TITLE
question about Tool Calling Distillation

### ISSUE THREAD

**Issue by mohamad-tohidi** [arcee-ai/DistillKit#46]
first of all i want to thank you for this amazing project.
i really like it and i will use it a lot.

my project, requires me to distill the teacher model into the student, in an Agentic enviornment.
meaning, my setup will call tools, and get outside informations.

in normal SFT BlackBox distillation that i have done before, this was not a problem, as i created the dataset in the standard format, and made sure that the tool results are masked so the student wont learn them.

in the distillation methods that you have in this project:
**do i have to create an enviornment for the teacher to use my tools online?**
**or i can add my dataset which has user input and the teachers outputs (without the logits) and we will just add the logits to each token of the teachers response that i already have**

thank you in advance for the time you spend reading my question.

**Comment by mohamad-tohidi** [comment]
i want to use the logits distillation by the way.

but i think the answer is the same for the logits and hidden state
